_rules/aria-required-accessible-name-gp8n89.md

```diff
@@ -0,0 +1,268 @@
+
+```html
+<div role="img" aria-label="Rating: 5 out of 5 stars">
+  <span aria-hidden="true">★★★★★</span>
+</div>
+```
+
+#### Passed Example 4
+
+The `div` element with role `checkbox` has an [accessible name][] provided thanks to the `aria-labelledby` attribute.
+
+```html
+<div role="checkbox" aria-checked="false" tabindex="0" aria-labelledby="pass-agree-tc"></div>
+<div id="pass-agree-tc">I agree with terms and conditions</div>
+```
+
+#### Passed Example 5
+
+The `div` element with role `dialog` has an [accessible name][] provided by the `aria-labelledby` attribute.
+
+```html
+<div role="dialog" aria-modal="true" tabindex="-1" aria-labelledby="pass-terms">
+  <h1 id="pass-terms">Terms</h1>
+  <p>These are our terms.</p>
+</div>
+```
+
+#### Passed Example 6
+
+The `div` element with role `heading` has an [accessible name][] provided by its content.
+
+```html
+  <div role="heading" aria-level="1">Terms</div>
+```
+
+#### Passed Example 7
+
+This `div` element with role `heading` is not [visible][], but is still [included in the accessibility tree][]. It has a non-empty [accessible name][] provided by its content.
+
+```html
+<div role="heading" aria-level="1" style="position: absolute; left: -9999px">ACT rules</div>
+```
+
+#### Passed Example 8
+
+The `div` element with role `link` has an [accessible name][] provided by its content.
+
+```html
+<div role="link" tabindex="0" onclick="location.href='https://act-rules.github.io/'">ACT Rules</div>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The `div` element with role `button` doesn't have an [accessible name][].
+
+```html
+<div role="button"></div>
+```
+
+#### Failed Example 2
+
+This `button` element with an [implicit semantic role] of `button` doesn't have an [accessible name][].
+
+```html
+<button></button>
+```
+
+#### Failed Example 3
+
+The `div` element with role `img` doesn't have an [accessible name][].
+
+```html
+<div role="img">
+  <span aria-hidden="true">★★★★★</span>
+</div>
+```
+
+#### Failed Example 4
+
+The `div` element with role `checkbox` doesn't have an [accessible name][].
+
+```html
+<div role="checkbox" aria-checked="false" tabindex="0"></div>
+<div>I agree with terms and conditions</div>
+```
+
+#### Failed Example 5
+
+The `dialog` role accepts an [accessible name][] only from the author and not from its content. Therefore, the `div` element with role `dialog` doesn't have an [accessible name][].
+
+```html
+<div role="dialog" aria-modal="true" tabindex="-1">
+  <h1>Terms</h1>
+  <p>These are our terms.</p>
+</div>
+```
+
+#### Failed Example 6
+
+The `div` element with role `heading` has an empty [accessible name][] due to the `display: none` CSS property set to its content.
+
+```html
+  <div role="heading" aria-level="1"><span style="display: none">Terms</span></div>
+```
+
+#### Failed Example 7
+
+The `button` element has an empty [accessible name][] given by its `aria-labelledby` attribute.
```

Comment:
I know it's mentioned in the background, but do we need to have a failed example where all major browsers ignore the accessible name computation rules and do provide the button with an accessible name?